mempool/v0/clist_mempool.go

```diff
@@ -613,6 +614,15 @@ func (mem *CListMempool) Update(
 		// https://github.com/tendermint/tendermint/issues/3322.
 		if e, ok := mem.txsMap.Load(tx.Key()); ok {
 			mem.removeTx(tx, e.(*clist.CElement), false)
+			// see if the transaction is a malleated transaction of a some parent
+			// transaction that exists in the mempool
+		} else if malleatedTx, isMalleated := types.UnwrapMalleatedTx(tx); isMalleated {
+			var parentKey [types.TxKeySize]byte
+			copy(parentKey[:], malleatedTx.OriginalTxHash)
+			err := mem.RemoveTxByKey(parentKey)
```

Comment:
[Not sure about this, but potentially important]
If we are removing the "originalTx" from the mempool itself, shouldn't we also add it to the cache (as we do with "non-malleable" transactions above)?